### PR TITLE
Openid Connect Core support - Round 2

### PR DESCRIPTION
**Pull request by thinkwelltwd**
I would like to use DOT, but need oidc support so I rebased [shaun's](https://github.com/shauns/django-oauth-toolkit/tree/oidc-in-action) PR onto [wiliamsouza's](https://github.com/wiliamsouza/django-oauth-toolkit/tree/openid-connect) PR and then rebased that onto master.

If it's not kosher to open another PR with other people's work, please let me know and I'll delete it. Otherwise, I would be very glad to get oidc support in this library.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -26,13 +27,15 @@ deps =
...
 	oauthlib>=3.1.0
+	jwcrypto
 	coverage
 	pytest
 	pytest-cov
 	pytest-django
 	pytest-xdist
 	py27: mock
 	requests
+	jwcrypto
```
Comment (fvlima) [inline tox.ini:38]: the jwcrypto is twice here

Comment (thinkwelltwd) [reply tox.ini]: Oops; yanked the duplicated.

File: oauth2_provider/migrations/0005_accesstoken_id_token.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from django.conf import settings
```
Comment (auvipy) [inline oauth2_provider/migrations/0005_accesstoken_id_token.py:1]: would be nice if you could remove extra migrations files and regen them

**Review by auvipy — CHANGES_REQUESTED** [review]

File: oauth2_provider/migrations/0005_accesstoken_id_token.py (hunk 2 above)
Comment (thinkwelltwd) [reply oauth2_provider/migrations/0005_accesstoken_id_token.py]: Done...

**Review by auvipy — APPROVED** [review]

**Comment by pawalt** [comment]
Hey there! Thanks for this PR; my team is very excited about getting started with it. Are there any examples or docs we could follow to get started?

Happy to move to an issue as well if comments on closed PRs aren't permitted.

**Comment by n2ygk** [comment]
@auvipy you've merged a significant bunch of changes but didn't request that the checklist be followed and @thinkwelltwd I'm wondering why this checklist is not in the PR description as it must have been templated there. Did you (inadvertently) remove it?:
```
- [ ] PR only contains one change (considered splitting up PR)
- [ ] unit-test added
- [ ] documentation updated
- [ ] `CHANGELOG.md` updated (only for user relevant changes)
- [ ] author name in `AUTHORS`
```

Specifically I see a number of problems that will require a follow-up PR @thinkwelltwd:
- [x] PR only contains one change (considered splitting up PR)
         I guess it's too late to confirm if this is the case.
- [x] unit-test added
        I see a bunch of tests are added, and coverage has decreased a little bit, so hopefully the added tests are enough.
- [ ] documentation updated
        I see no documentation of the new OIDC features. Please document. All this great work is worth nothing to other users if they can't see that it's there. We don't merge code just for personal use.
- [ ] `CHANGELOG.md` updated (only for user relevant changes)
       This is major new functionality, requiring a major release per semver yet there's nothing about it in the CHANGELOG.
- [ ] author name in `AUTHORS`
       @thinkwelltwd please add your name. Credit where credit is due along with those of @shauns and @wiliamsouza 

Would @thinkwelltwd and/or @auvipy please submit a documentation PR to remediate this issue? I'm reluctant to push a new release with the master branch in the state it is right now.

@auvipy I'm kind of disappointed as we've discussed this issue of merging before ready before. This lack of process would someone considering using this library to think twice about the quality of it.

**Comment by n2ygk** [comment]
This will need to be re-opened/recreated as a new PR.

**Pull request merged**